### PR TITLE
(PUP-8347) usr error_location for auth config parsing

### PR DESCRIPTION
**Pull request by ericldelaney**
Use the error_location() method when parsing network auth config files

### CODE REVIEW

**Comment by ericldelaney** [comment]
jenkins please test this

**Comment by ericldelaney** [comment]
jenkins please test this

File: lib/puppet/network/auth_config_parser.rb [hunk 1]
```diff
@@ -29,11 +29,14 @@ def parse_rights
...
           #TRANSLATORS "path" is a configuration file entry and should not be translated
-          raise Puppet::ConfigurationError, _("Missing or invalid 'path' before right directive at line %{count} of %{file}") % { count: count, file: @file }
+          raise Puppet::ConfigurationError, _("Missing or invalid 'path' before right directive at %{error_location}") %
+              { error_location: Puppet::Util::Errors.error_location(@file, count) }
         end
         parse_right_directive(right, $1, $2, count)
       else
-        raise Puppet::ConfigurationError, _("Invalid line %{count}: %{line}") % { count: count, line: line }
+        error_location_str = Puppet::Util::Errors.error_location(nil, count)
+        raise Puppet::ConfigurationError, _("Invalid entry at %{error_location}: %{file_text}") %
```
Comment (ericldelaney) [inline lib/puppet/network/auth_config_parser.rb]: Changed this invalid entry error string

File: spec/integration/network/authconfig_spec.rb [hunk 2]
```diff
@@ -97,13 +97,13 @@ def request(args = {})
     it 'should warn about missing path before allow_ip in stanza' do
       expect {
         add_raw_stanza("allow_ip 10.0.0.1\n")
-      }.to raise_error Puppet::ConfigurationError, /Missing or invalid 'path' before right directive at line.*/
+      }.to raise_error Puppet::ConfigurationError, /Missing or invalid 'path' before right directive at \(line: .*/
```
Comment (jtappa) [inline spec/integration/network/authconfig_spec.rb]: missing closing paren? and at 106 too?

Comment (ericldelaney) [reply spec/integration/network/authconfig_spec.rb]: sure

**Comment by ericldelaney** [comment]
@jtappa Updated

File: lib/puppet/network/auth_config_parser.rb [hunk 3]
```diff
@@ -65,8 +68,9 @@ def parse_right_directive(right, var, value, count)
     when /auth(?:enticated)?/
       modify_right(right, :restrict_authenticated, value, _("adding authentication %{value}"), count)
     else
-      raise Puppet::ConfigurationError,
-        _("Invalid argument '%{var}' at line %{count}") % { var: var, count: count }
+      error_location_str = Puppet::Util::Errors.error_location(nil, count)
```
[the remark this answers is not in the document]
Comment (jtappa) [reply lib/puppet/network/auth_config_parser.rb]: Maybe merits some sort of maint ticket or trello card for team discussion?

[the remark this answers is not in the document]
Comment (jtappa) [reply lib/puppet/network/auth_config_parser.rb]: (I am +1000 for standardization, consistency is my BFF)

[the remark this answers is not in the document]
Comment (ericldelaney) [reply lib/puppet/network/auth_config_parser.rb]: Ok let me get all these PR's merged and I'll make sure they're all consistent

[the remark this answers is not in the document]
Comment (jhelwig) [reply lib/puppet/network/auth_config_parser.rb]: If we didn't have to worry about Ruby 1.9 support (I know why, but...whyyyyyy?), I'd say this method is a great example of when to use [keyword arguments](https://robots.thoughtbot.com/ruby-2-keyword-arguments).

**Pull request merged**